src/streams.h

```diff
@@ -530,6 +530,14 @@ class AutoFile
     //
     // Stream subset
     //
+    void seek(int64_t offset, int origin)
+    {
+        if (!file)
+            throw std::ios_base::failure("CAutoFile::read: file handle is nullptr");
+        if (fseek(file, offset, origin) != 0)
+            throw std::ios_base::failure(feof(file) ? "CAutoFile::seek: end of file" : "CAutoFile::seek: fseek failed");
```

Comment:
1d093bbf97b8d4ec5413ca560ac79e56f23d7f1d: maybe log the exact error code.

Reply:
It's kind of annoying to do that, so I'll leave it as is. The other functions do the same.